source/model-generator/TaintConfigTemplate.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <mariana-trench/model-generator/ModelTemplates.h>
+#include <mariana-trench/model-generator/TaintConfigTemplate.h>
+
+namespace marianatrench {
+
+TaintConfigTemplate TaintConfigTemplate::from_json(const Json::Value& value, Context& context) {
+  TaintConfig taint_config = TaintConfig::from_json(value, context);
+  std::vector<AnnotationFeatureTemplate> annotation_features;
+  if (value.isMember("via_annotation")) {
+    for (const auto& feature_value : JsonValidation::null_or_array(value["via_annotation"])) {
+      annotation_features.emplace_back(AnnotationFeatureTemplate::from_json(feature_value));
+    }
+  }
+  return TaintConfigTemplate{std::move(taint_config), std::move(annotation_features)};
+}
+
+TaintConfig TaintConfigTemplate::instantiate(const Method* method, Context& context) const {
+  return instantiate(method, context, TemplateVariableMapping{});
+}
+
+TaintConfig TaintConfigTemplate::instantiate(const Method* method, Context& context, const TemplateVariableMapping& parameter_positions) const {
+  FeatureSet new_user_features;
+  for (const auto& annotation_feature : annotation_features_) {
+    const Feature* MT_NULLABLE new_feature = annotation_feature.instantiate(method, context, parameter_positions);
+    if (new_feature) {
+      new_user_features.add(new_feature);
+    }
+  }
+
+  TaintConfig taint_config{taint_config_};
+  for (auto feature : new_user_features) {
+    taint_config.add_user_feature(feature);
+  }
```

Comment:
Instead of iterating, you can add a method that takes `FeatureSet` and join
```
TaintConfig::add_user_feature_set(FeatureSet feature_set) {
  user_features_.join_with(feature_set);
}
```